02-esc/index.js

```diff
@@ -1,7 +1,23 @@
 import { getDataFromAPI } from "./getDataFromAPI.js";
+import { showAllChallenges } from "./showAllChallenges.js";
 
-// getDataFromAPI returns an array with all the API data
-// console.log(await getDataFromAPI());
+const challenge_list = document.querySelector('.challenge-list');
+
+let resFromAPI = await getDataFromAPI();
+
+function sortRatingArray(resFromAPI){ 
```

Comment:
The name of the `resFromAPI` parameter indicates to me that maybe you are not yet comfortable with thinking about functions as vehicles for separation of concerns.

This function has nothing to do with the API. It doesn't really care where the list of challenges comes from, as long as it has a certain shape (challenges need to have a `rating` property, and be passable to  `showAllChallenges()`).

Inside this function, the source is irrelevant, and the name of the parameter is only used inside of this function, so why not call it something simpler, e.g. simply `challenges` instead of `resFromAPI`.

This also helps to separate, in the mind of the reader of your code, the `resFromAPI` name in this function from the `resFromAPI` name outside of the function. When you declare a variable (like here) inside a function, it's name will override for the scope of the function, the name of any variables declared outside.

This means that although it looks like `resFromAPI` is the same thing on line 6 and 10, that's not guaranteed. They are only the same thing if you pass `resFromAPI` from the outer scope to the inner scope as a parameter on line 20.

Imagine changing line 20 to `sortRatingArray([])`. Now you're passing an empty array, which means that the `resFromAPI´ variable inside the function will be an empty array, while the `resFromAPI` variable outside of the function is something else.

To summarize: Avoid reusing the same name in two different scopes that are accessible to each other. It's legal, but can cause confusion, and often it doesn't make the most sense anyway since the two scopes probably have different perspective on the variable anyway.